xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/flamingo.js

```diff
@@ -40,14 +40,82 @@ require(['jquery', 'bootstrap'], function($) {
+   ** The content to display inside.
+  In addition, the drawer can contain a `.drawer-close` button that will close the drawer on click.
+  For an example of drawer creation, see #tmDrawerActivator and #tmDrawer.
+ */
+require(['jquery'], function($) {
+  $(document).ready(function() {
+    $('.drawer-opener').each(function (index) {
+      // Setting up the drawer.
+      let drawerOpener = $(this);
+      let drawerId = drawerOpener.attr('aria-controls');
+      let drawerContainer = $(document.getElementById(drawerId));
+
+      let openDrawer = () => drawerContainer.trigger('drawer' + index + '.opened');
+      let closeDrawer = () => drawerContainer.trigger('drawer' + index + '.closed');
+      drawerOpener.on('click', openDrawer);
+      // Close drawer when clicking the backdrop (or any element outside of the drawer itself).
+      drawerContainer.on('click', (event) => {
+        let drawerzone = event.target.getBoundingClientRect();
+        if (event.target === drawerContainer.get(0) &&
+            (drawerzone.left > event.clientX || drawerzone.right < event.clientX
+            || drawerzone.top > event.clientY || drawerzone.bottom < event.clientY)) {
+          closeDrawer();
+        }
+      });
+      // Close drawer when clicking on a close button inside it
+      drawerContainer.find('.drawer-close').on('click', closeDrawer);
+      
+      drawerContainer.on('drawer' + index + '.opened', function (event) {
+        // We use the drawer-transitioning class to make sure the transition to slidein is not shortcutted when showing the modal
+        drawerContainer.addClass('drawer-transitioning');
+        drawerOpener.attr('aria-expanded', 'true');
+        drawerContainer.get(0).showModal();
+        drawerContainer.removeClass('drawer-transitioning');
+        // The drawer can be closed by pressing the ESC key
+        $("body").on('keydown.drawer' + index + 'Close', function (event) {
+          if (event.key === 'Escape') {
+            closeDrawer();
+          }
+        });
+      }).on('drawer' + index + '.closed', function (event) {
+        drawerOpener.attr('aria-expanded', 'false');
+
+        function waitTransition() {
+          drawerContainer.get(0).close();
+          drawerContainer.removeClass('drawer-transitioning');
+          drawerContainer.get(0).removeEventListener('transitionend', waitTransition);
+        }
+
+        // We use the drawer-transitioning class to give time for the hide modal transition to play.
+        drawerContainer.addClass('drawer-transitioning');
+        drawerContainer.get(0).addEventListener('transitionend', waitTransition);
+        // We remove the listener that was created when the drawer opened up
+        $("body").off('keydown.drawer' + index + 'Close');
+      });
+
+      // When the drawer is closed, collapse sub items
+      $(body).on('drawer' + index + '.closed', function () {
+        $('.drawer-menu-sub-item').removeClass('in').attr('aria-expanded', 'false');
```

Comment:
This seems wrong in the context of multiple drawers as this would remove the `in` class in all drawers whenever any drawer is closed. Wouldn't it be better to move that into the previous event handler and to only collapse sub items of that drawer?

Reply:
Addressed in https://github.com/xwiki/xwiki-platform/pull/2282/commits/dead0799794baaa46fe4ae9460c55e923d745b61 :+1: 
Leaving this listener on 'body' was important when the focus could leave the drawer, but as of now, it's trapped inside the container, so it's okay to have this listener on the container directly. Changing this scope allows to close items in one drawer only.

This solution does not work properly when drawer containers are nested. I think this is an okay tradeoff, AFAIK they shouldn't be. I still added a couple lines in the doc to indicate that :) https://github.com/xwiki/xwiki-platform/pull/2282/commits/2494a8e9f67ec1ea959e069b59e7a24f1026c18f

Reply:
I'm not so sure about the nesting, but when they are nested, I would expect that outer drawers are always open when inner drawers are opened so in this case, there really shouldn't be any problem because of this.